generator/generators/RustGenerator.py

```diff
@@ -0,0 +1,76 @@
+'''
+ Copyright (c) 2018 HERE Europe B.V.
+ See the LICENSE file in the root of this project for license details.
+'''
+
+from generator.tree.nodes.resources import (Vector, Multivector, Instance, RawData, BoundResource,
+    Archive as ArchiveResource)
+from generator.tree.nodes.trivial import Structure, Constant, Enumeration
+from generator.tree.nodes.archive import Archive
+from generator.tree.syntax_tree import SyntaxTree
+from .BaseGenerator import BaseGenerator
+
+import re
+
+class RustGenerator(BaseGenerator):
+
+    RESERVED_KEYWORDS = [
+        "abstract", "alignof", "as", "become", "box", "break", "const", "continue", "crate", "do",
+        "else", "enum", "extern", "false", "final", "fn", "for", "if", "impl", "in", "let", "loop",
+        "macro", "match", "mod", "move", "mut", "offsetof", "override", "priv", "proc", "pub",
+        "pure", "ref", "return", "self", "sizeof", "static", "struct", "super", "trait", "true",
+        "type", "typeof", "unsafe", "unsized", "use", "virtual", "where", "while", "yield"]
+
+    def __init__(self):
+        BaseGenerator.__init__(self, "rust/rust.jinja2")
+
+    def _supported_nodes(self):
+        return [Structure, Archive, Constant, Enumeration]
+
+    def _populate_environment(self, env):
+        def _camel_to_snake_case(s):
+            s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', s)
+            return re.sub('([a-z0-9])(A-Z)', r'\1_\2', s1).lower()
+
+        env.filters["camel_to_snake_case"] = _camel_to_snake_case
+
+        def _snake_to_upper_camel_case(s):
+            return ''.join(p.title() for p in  s.split('_'))
+
+        env.filters["snake_to_upper_camel_case"] = _snake_to_upper_camel_case
+
+        def _rust_doc(s):
+            lines = [
+                re.sub(r'^[ \t]*(/\*\*|/\*|\*/|\*)\s*(.*?)\s*(\*/)?$', r"/// \2", line).strip()
+                for line in s.split('\n')
+            ]
+            start = 0
+            end = len(lines)
+            if lines[0] == "///":
+                start = 1
+            if lines[-1] == "///":
+                end = -1;
```

Comment:
I am not sure what intended here, but maybe in a same fashion as previous `if` clause it should be `-2`?